backend/video/views.py

```diff
@@ -184,74 +196,63 @@ def get_video(request):
             .first()
         )
 
-        # If it does, return the existing transcript
-        if transcript:
-
-            serializer = VideoSerializer(video)
-
-            # Check if it's audio only
-            if audio_only:
-                return Response(
-                    {
-                        "direct_audio_url": direct_audio_url,
-                        "subtitles": subtitles_list,
-                        "video": serializer.data,
-                        "transcript_id": transcript.id,
-                    },
-                    status=status.HTTP_200_OK,
-                )
-
-            return Response(
-                {
-                    "direct_video_url": direct_video_url,
-                    "subtitles": subtitles_list,
-                    "video": serializer.data,
-                    "transcript_id": transcript.id,
-                },
-                status=status.HTTP_200_OK,
-            )
+        if not transcript:
 
-        # Save a transcript object
-        transcript_obj = Transcript(
-            transcript_type=ORIGINAL_SOURCE,
-            video=video,
-            language=lang,
-            payload=video.subtitles,
-        )
-        transcript_obj.save()
+            # Save a transcript object
+            transcript = Transcript(
+                transcript_type=ORIGINAL_SOURCE,
+                video=video,
+                language=lang,
+                payload=video.subtitles,
+            )
+            transcript.save()
 
-        serializer = VideoSerializer(video)
-        return Response(
-            {
-                "direct_audio_url": direct_audio_url,
-                "direct_video_url": direct_video_url,
-                "subtitles": subtitles_list,
-                "video": serializer.data,
-                "transcript_id": transcript_obj.id,
-            },
-            status=status.HTTP_200_OK,
-        )
+        # Add the transcript to the response data
+        response_data["transcript_id"] = transcript.id
 
+    # Check if it's audio only
+    if audio_only:
+        response_data["audio_url"] = direct_audio_url
     else:
-        serializer = VideoSerializer(video)
-        return Response(
-            {
-                "direct_audio_url": direct_audio_url,
-                "direct_video_url": direct_video_url,
-                "subtitles": subtitles_list,
-                "video": serializer.data,
-            },
-            status=status.HTTP_200_OK,
-        )
+        response_data["video_url"] = direct_video_url
+
+    return Response(
+        response_data,
+        status=status.HTTP_200_OK,
+    )
 
 
+@swagger_auto_schema(
+    method="get",
+    manual_parameters=[
+        openapi.Parameter(
+            "audio_only",
```

Comment:
If we are fixing with `is_audio_only`, please modify here as well.

As well as in the `get_video()` endpoint.

Reply:
Updated